packages/use-intl/test/useTranslations.test.tsx

```diff
@@ -234,6 +234,35 @@ it('has a stable reference', () => {
   screen.getByText('2');
 });
 
+it('caches by locale, namespace and key', () => {
```

Comment:
I think the caching should be a implementation detail here.

```suggestion
it('renders the correct message when the namespace changes', () => {
```